pygfx/utils/load.py

```diff
@@ -5,72 +5,432 @@
+    is mesh-only (e.g. `.stl`) we will instead construct a minimal scene.
+
+    This function requires the trimesh library.
+
+    Parameters
+    ----------
+    path : str
+        The filepath. Can be a local file or a URL.
+    flatten : bool
+        If True, will ignore any hierarchical structure in the scene graph and
+        instead import as a flat list of objects.
+    meshes : bool
+        Whether to load meshes.
+    materials : bool
+        Whether to load materials.
+    lights : bool, optional
+        Whether to load lights:
+          - True (default): lights are imported if present but if not, minimal lights are added
+          - False: ignore any lights in the file but add minimal lights to scene
+          - None: no lights are loaded
+    camera : bool, optional
+        Whether to load camera:
+          - True (default): camera is imported if present but if not, a generic camera is added
+          - False: ignore any camera in the file but add a camera to scene
+          - None: no camera is loaded
+    remote_ok : bool
+        Whether to allow loading files from URLs, by default False.
+
+    Returns
+    -------
+    scene : pygfx.Scene
+        The loaded scene.
+
+    See Also
+    --------
+    load_mesh
+        Returns the flat meshes contained in a file.
+
+    """
     import trimesh  # noqa
 
-    scene = trimesh.load(path)
+    # Trimesh's load() performs a similar check and refers
+    # loading from URLs to load_remote()
+    if "https://" in str(path) or "http://" in str(path):
+        if not remote_ok:
+            raise ValueError(
+                "Loading scenes from URLs is disabled. "
+                "Set remote_ok=True to allow loading from URLs."
+            )
+        tm_scene = trimesh.load_remote(path)
+    else:
+        tm_scene = trimesh.load(path)
+
+    return scene_from_trimesh(
+        tm_scene,
+        flatten=flatten,
+        meshes=meshes,
+        materials=materials,
+        lights=lights,
+        camera=camera,
+    )
+
+
+def meshes_from_trimesh(scene, materials=True, apply_transforms=True):
+    """Converts a trimesh scene into a flat list of pygfx Mesh objects.
+
+    Parameters
+    ----------
+    scene : trimesh.Scene
+        The scene to convert.
+    materials : bool
+        Whether to import materials. If False, a default material will be created.
+    apply_transforms : bool
+        Whether to apply the scene graph transforms directly to the meshes, by default True.
+
+    Returns
+    -------
+    meshes : list
+        A list of loaded meshes.
+
+    """
+    import trimesh  # noqa
+
+    # If this is a single object we can just convert and return it
     if isinstance(scene, trimesh.Trimesh):
         m = gfx.Mesh(
             gfx.geometry_from_trimesh(scene),
-            gfx.MeshPhongMaterial(),
+            gfx.material_from_trimesh(scene.visual.material),
```

Comment:
This triggers an exception for the docs CI (which use trimesh 4.2.4):
```
../examples/feature_demo/show_stl.py unexpectedly failed to execute correctly:

    Traceback (most recent call last):
      File "/home/docs/checkouts/readthedocs.org/user_builds/pygfx/checkouts/707/examples/feature_demo/show_stl.py", line 36, in <module>
        meshes = gfx.load_mesh(model_dir / "teapot.stl")
      File "/home/docs/checkouts/readthedocs.org/user_builds/pygfx/envs/707/lib/python3.9/site-packages/pygfx/utils/load.py", line 65, in load_mesh
        return meshes_from_trimesh(scene, apply_transforms=True)
      File "/home/docs/checkouts/readthedocs.org/user_builds/pygfx/envs/707/lib/python3.9/site-packages/pygfx/utils/load.py", line 168, in meshes_from_trimesh
        gfx.material_from_trimesh(scene.visual.material),
    AttributeError: 'ColorVisuals' object has no attribute 'material'
```

Reply:
Whoops. It turns out `ColorVisuals` are used for vertex and face colors but they can also be empty. These visuals do not have a `.material` but we still need to translate them into a `pygfx` material. I fixed this with 
be5be00 and 7fe732 and that example now completes. 

Since `ColorVisual` is now explicitly converted, the aesthetics also changed slightly because trimesh uses fairly high default values for shininess:

<img width="752" alt="Screenshot 2024-04-05 at 10 30 11" src="https://github.com/pygfx/pygfx/assets/7161148/c5ea31fa-99f1-400b-b7a8-683873be912b">

Not sure if you care about that?